### ISSUE TITLE
Weird characters when loading tidyverse

### ISSUE THREAD

**Issue by harrismcgehee** [tidyverse/reprex#150]
Do you recommend `suppressStartupMessages()` for `library("tidyverse")` + `reprex::reprex()`?

Do I need to update a package?

``` r
library("tidyverse")
#> â”€â”€ Attaching packages â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ tidyverse 1.2.1 â”€â”€
#> âœ” ggplot2 2.2.1.9000     âœ” purrr   0.2.4     
#> âœ” tibble  1.3.4          âœ” dplyr   0.7.4     
#> âœ” tidyr   0.7.2          âœ” stringr 1.2.0     
#> âœ” readr   1.1.1          âœ” forcats 0.2.0
#> â”€â”€ Conflicts â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ tidyverse_conflicts() â”€â”€
#> âœ– dplyr::filter() masks stats::filter()
#> âœ– dplyr::lag()    masks stats::lag()
```

<details><summary>Session info</summary>

``` r
devtools::session_info()
#> Session info --------------------------------------------------------------
#>  setting  value                       
#>  version  R version 3.3.2 (2016-10-31)
#>  system   x86_64, linux-gnu           
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  tz       America/New_York            
#>  date     2018-01-03
#> Packages ------------------------------------------------------------------
#>  package    * version    date       source                            
#>  assertthat   0.2.0      2017-04-11 cran (@0.2.0)                     
#>  backports    1.0.5      2017-01-18 CRAN (R 3.3.2)                    
#>  bindr        0.1        2016-11-13 cran (@0.1)                       
#>  bindrcpp     0.2        2017-06-17 CRAN (R 3.3.2)                    
#>  broom        0.4.2      2017-02-13 CRAN (R 3.3.2)                    
#>  cellranger   1.1.0      2016-07-27 CRAN (R 3.3.2)                    
#>  cli          1.0.0      2017-11-05 CRAN (R 3.3.2)                    
#>  colorspace   1.3-2      2016-12-14 CRAN (R 3.3.2)                    
#>  crayon       1.3.4      2017-09-16 CRAN (R 3.3.2)                    
#>  devtools     1.12.0     2016-12-05 CRAN (R 3.3.2)                    
#>  digest       0.6.12     2017-01-27 CRAN (R 3.3.2)                    
#>  dplyr      * 0.7.4      2017-09-28 CRAN (R 3.3.2)                    
#>  evaluate     0.10.1     2017-06-24 cran (@0.10.1)                    
#>  forcats    * 0.2.0      2017-01-23 CRAN (R 3.3.2)                    
#>  foreign      0.8-67     2016-09-13 CRAN (R 3.3.2)                    
#>  ggplot2    * 2.2.1.9000 2017-11-14 Github (tidyverse/ggplot2@41f154f)
#>  glue         1.2.0      2017-10-29 CRAN (R 3.3.2)                    
#>  gtable       0.2.0      2016-02-26 CRAN (R 3.3.2)                    
#>  haven        1.1.0      2017-07-09 CRAN (R 3.3.2)                    
#>  hms          0.3        2016-11-22 CRAN (R 3.3.2)                    
#>  htmltools    0.3.6      2017-04-28 cran (@0.3.6)                     
#>  httr         1.3.1      2017-08-20 CRAN (R 3.3.2)                    
#>  jsonlite     1.5        2017-06-01 cran (@1.5)                       
#>  knitr        1.17       2017-08-10 cran (@1.17)                      
#>  lattice      0.20-35    2017-03-25 CRAN (R 3.3.2)                    
#>  lazyeval     0.2.1      2017-10-29 cran (@0.2.1)                     
#>  lubridate    1.7.1      2017-11-03 CRAN (R 3.3.2)                    
#>  magrittr     1.5        2014-11-22 CRAN (R 3.3.2)                    
#>  memoise      1.0.0      2016-01-29 CRAN (R 3.3.2)                    
#>  mnormt       1.5-5      2016-10-15 CRAN (R 3.3.2)                    
#>  modelr       0.1.1      2017-07-24 CRAN (R 3.3.2)                    
#>  munsell      0.4.3      2016-02-13 CRAN (R 3.3.2)                    
#>  nlme         3.1-131    2017-02-06 CRAN (R 3.3.2)                    
#>  pkgconfig    2.0.1      2017-03-21 cran (@2.0.1)                     
#>  plyr         1.8.4      2016-06-08 CRAN (R 3.3.2)                    
#>  psych        1.6.12     2017-01-08 CRAN (R 3.3.2)                    
#>  purrr      * 0.2.4      2017-10-18 CRAN (R 3.3.2)                    
#>  R6           2.2.2      2017-06-17 cran (@2.2.2)                     
#>  Rcpp         0.12.13    2017-09-28 cran (@0.12.13)                   
#>  readr      * 1.1.1      2017-05-16 CRAN (R 3.3.2)                    
#>  readxl       1.0.0      2017-04-18 CRAN (R 3.3.2)                    
#>  reshape2     1.4.2      2016-10-22 CRAN (R 3.3.2)                    
#>  rlang        0.1.4      2017-11-05 CRAN (R 3.3.2)                    
#>  rmarkdown    1.6        2017-06-15 cran (@1.6)                       
#>  rprojroot    1.2        2017-01-16 CRAN (R 3.3.2)                    
#>  rstudioapi   0.7        2017-09-07 cran (@0.7)                       
#>  rvest        0.3.2      2016-06-17 CRAN (R 3.3.2)                    
#>  scales       0.5.0.9000 2017-10-09 Github (hadley/scales@d767915)    
#>  stringi      1.1.5      2017-04-07 cran (@1.1.5)                     
#>  stringr    * 1.2.0      2017-02-18 CRAN (R 3.3.2)                    
#>  tibble     * 1.3.4      2017-08-22 cran (@1.3.4)                     
#>  tidyr      * 0.7.2      2017-10-16 CRAN (R 3.3.2)                    
#>  tidyverse  * 1.2.1      2017-11-14 CRAN (R 3.3.2)                    
#>  withr        2.1.0.9000 2017-11-14 Github (jimhester/withr@a9ebeb3)  
#>  xml2         1.1.1      2017-01-24 CRAN (R 3.3.2)                    
#>  yaml         2.1.14     2016-11-12 CRAN (R 3.3.2)
```

</details>

**Comment by jennybc** [comment]
@gaborcsardi provides this diagnosis:

> locale is set to UTF-8, but the terminal cannot handle that.
> I am not sure what `ui: X11` is on Linux

What version of reprex do you have? If you're willing to run dev version from GitHub, it will suppress the tidyverse startup message by default. But that doesn't solve your fundamental problem -- which btw would not be limited to reprex. I think reprex is just exposing a problem.

**Comment by gaborcsardi** [comment]
Btw. you can turn off the unicode characters with 
```
options(cli.unicode = FALSE)
```

**Comment by harrismcgehee** [comment]
@jennybc I have reprex version 0.1.1.

I'm not sure if I understand what you both are saying. Are you suggesting just always turn off the unicode characters?

The console looks good in RStudio Server; just not in these reprex .md files.

![image](https://user-images.githubusercontent.com/9814456/34544817-95c3f2a0-f0b6-11e7-9dda-af417f3cd663.png)

**Comment by harrismcgehee** [comment]
@gaborcsardi 

With your options, we get almost there except the checkmark character?

``` r
options(cli.unicode = FALSE)
library("tidyverse")
#> -- Attaching packages ------------------------------------------------------------------------------------------------------------------------------------------------- tidyverse 1.2.1 --
#> âˆš ggplot2 2.2.1.9000     âˆš purrr   0.2.4     
#> âˆš tibble  1.3.4          âˆš dplyr   0.7.4     
#> âˆš tidyr   0.7.2          âˆš stringr 1.2.0     
#> âˆš readr   1.1.1          âˆš forcats 0.2.0
#> -- Conflicts ---------------------------------------------------------------------------------------------------------------------------------------------------- tidyverse_conflicts() --
#> x dplyr::filter() masks stats::filter()
#> x dplyr::lag()    masks stats::lag()
```

**Comment by gaborcsardi** [comment]
AFAIR the checkmark character is a `tidyverse` bug, probably fixes in the GH version as well.

Oh, so this is about the md files. Then maybe the editor or viewer you open the md file with does not support UTF-8.

**Comment by jennybc** [comment]
If you install reprex from GitHub (`devtools::install_github("tidyverse/reprex")`), there is a new argument `tidyverse_quiet` which defaults to ... an option, which defaults to ... `TRUE`. The effect is to suppress the inclusion of the tidyverse startup message.

But that is just masking a problem you should get to the bottom of.

It's hard to see how this is specific to reprex. If you make a new `.R` script with this content:

``` r
#' ---
#' title: "Encoding Joy"
#' output: github_document
#' ---

library(tidyverse)
```

and use the "Compile Report" button or call `rmarkdown::render()` on it, how does the `.md` look?

Can you explain exactly how this `.md` got into the issue above? I'm trying to understand what all is has passed through.

**Comment by harrismcgehee** [comment]
re: where the .md came from:
When I use the `reprex` addin, it tries to copy the code to the clipboard but fails on RStudio Server. However, it gives the `* For now, see the temp file: <temp>.md` .  That is the `.md` that I am referencing and copying from.  
I open it after generating it with `file.show('<temp>.md')`.  It is in the resulting pop-up (in Chrome if it matters) that the checkmark character is mangled.

re: Compiling Report:
Aha! This method surfaced the issue (for me, probably not you two). When I use the Files pane to navigate to the compiled `.md` and open in the Source pane, RStudio is able to show the checkmarks.  When I just use `file.show` on the same compiled `.md`, I get the mangled characters.

I will stick to the GitHub and future CRANs of `reprex` with the `tidyverse_quiet = TRUE`! Super thanks to you both. Thank you for the wonderful package and wonderful GitHub support!

ps. I'm never sure if I should close issues or not in case there are lingering things package authors would like to address. Also, I hate to rob you of the satisfaction of marking something as done.

**Comment by jennybc** [comment]
@harrismcgehee Glad you got it sorted out. If you are going to be forced to manually copy reprex md to the clipboard, you might want to call the function yourself (vs using the addin) and default to `reprex(..., outfile = "NA)` or `reprex(..., outfile = "reprex")` to get things left behind in current working directory. I suspect your use of `file.show()` might be related to the files being left in a very inconvenient place (temp dir).

**Comment by jennybc** [comment]
@harrismcgehee Another useful tip from @jmcphers: use `file.edit()` instead of `file.show()` here. That would open the file in RStudio.

**Comment by EarlGlynn** [comment]
Thanks @jennybc for quick reply.  Does not involve reprex.  I was responding to the title and other comments about tidyverse and weird characters.

Here's my problem example:
    ---
    title: "Initial look at MIMIC-III Using RPostgres"
    output:
      html_notebook:
        toc: yes
      html_document:
        toc: yes
    ---

    ```{r setup, include=FALSE}
    knitr::opts_chunk$set(echo = TRUE)
    options(cli.unicode = FALSE)
    ```

    ```{r, comment=NA}
    library(DBI)
    library(RPostgres)
    library(tidyverse)
    library(dbplyr, warn.conflicts=FALSE)
    library(kableExtra)
    ```                                    
Knit to HTML works fine every time.  But this does not:  Restart R and Clear Output | Run All | Preview. This is what I see from the notebook.

    [30m-- [1mAttaching packages[22m --------------------------------------- tidyverse 1.2.1 --[39m
    [30m[32mv[30m [34mggplot2[30m 3.0.0     [32mv[30m [34mpurrr  [30m 0.2.5
    [32mv[30m [34mtibble [30m 1.4.2     [32mv[30m [34mdplyr  [30m 0.7.6
    [32mv[30m [34mtidyr  [30m 0.8.1     [32mv[30m [34mstringr[30m 1.3.1
    [32mv[30m [34mreadr  [30m 1.1.1     [32mv[30m [34mforcats[30m 0.3.0[39m
    [30m-- [1mConflicts[22m ------------------------------------------ tidyverse_conflicts() --
    [31mx[30m [34mdplyr[30m::[32mfilter()[30m masks [34mstats[30m::filter()
    [31mx[30m [34mdplyr[30m::[32mlag()[30m    masks [34mstats[30m::lag()[39m

I've seen similar problems in a number of notebooks at work, with "weird" characters like above.  Thanks for any help with this.

**Comment by EarlGlynn** [comment]
Thanks for your replies, @jennybc and @gaborcsardi. I'm using the latest RStudio, version 1.1.456 on a PC.  I'll open an IDE discussion about this.

"Preview" is the way of looking at an R Studio Notebook (File | New File | R Notebook) vs an R Markdown script.  The YAML header above (with proper indentation) let's me switch back and forth between the two.  For "Reproducible Research" I always knit the document.  For code/data exploration, when one can run cells in any order for experiments, one can "Preview" an HTML version of the output in an .nb.html file.  While not a necessarily a "reproducible" approach, a .nb.html can be viewed in a brower, but also opened in RStudio to work with the code more.  I knit my final run after the experimenting is over.

**Comment by gaborcsardi** [comment]
@EarlGlynn Got it, thanks. Your pre-amble code above is garbled up, I had to fix that to get a "Preview"  button. Yes, this seems like an IDE problem, although I am not entirely sure if it is supposed to handle colors in this case. 

Once workaround is to get rid of  the colors, in your setup chunk:

```
options(crayon.enabled = FALSE)
options(crayon.colors = 1)
crayon::num_colors(forget = TRUE)
```

**Comment by gsimchoni** [comment]
Having the same issue too specifically with Preview function for a R notebook, I have Rstudio Version 1.2.1335, and here is another person struggling with this on SO (with a few upvotes):

https://stackoverflow.com/questions/56492132/strange-characters-in-preview-of-r-notebook

```
> sessionInfo()
R version 3.6.1 (2019-07-05)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18362)

Matrix products: default

locale:
[1] LC_COLLATE=English_Israel.1252  LC_CTYPE=English_Israel.1252    LC_MONETARY=English_Israel.1252 LC_NUMERIC=C                   
[5] LC_TIME=English_Israel.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] lubridate_1.7.4       fivethirtyeight_0.5.0 forcats_0.4.0         stringr_1.4.0         dplyr_0.8.3           purrr_0.3.3          
 [7] readr_1.3.1           tidyr_1.0.0           tibble_2.1.3          ggplot2_3.2.1         tidyverse_1.2.1      

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.5 xfun_0.8         haven_2.1.1      lattice_0.20-38  colorspace_1.4-1 vctrs_0.2.0      generics_0.0.2   htmltools_0.3.6 
 [9] yaml_2.2.0       base64enc_0.1-3  utf8_1.1.4       rlang_0.4.1      pillar_1.4.2     glue_1.3.1       withr_2.1.2      modelr_0.1.5    
[17] readxl_1.3.1     lifecycle_0.1.0  munsell_0.5.0    gtable_0.3.0     cellranger_1.1.0 rvest_0.3.4      evaluate_0.14    knitr_1.23      
[25] fansi_0.4.0      broom_0.5.2      Rcpp_1.0.3       scales_1.0.0     backports_1.1.5  jsonlite_1.6     hms_0.5.2        packrat_0.5.0   
[33] digest_0.6.22    stringi_1.4.3    grid_3.6.1       cli_2.0.1        tools_3.6.1      magrittr_1.5     lazyeval_0.2.2   crayon_1.3.4    
[41] pkgconfig_2.0.3  zeallot_0.1.0    rsconnect_0.8.15 xml2_1.2.2       assertthat_0.2.1 rmarkdown_1.14   httr_1.4.1       rstudioapi_0.10 
[49] R6_2.4.1         nlme_3.1-140     compiler_3.6.1 
```

**Comment by gsimchoni** [comment]
Update: there's an issue for this here:
https://github.com/rstudio/rstudio/issues/2364

Apparently solved in Rstudio v1.3.
